### ISSUE TITLE
[flang][compiler-rt] incompatible names for clang_rt.builtins

### ISSUE THREAD

**Issue by h-vetinari** [llvm/llvm-project#95698]
On windows, flang expects to link whatever is returned by `getCompilerRTBasename`:
https://github.com/llvm/llvm-project/blob/630a6dd687ab4ec2038fc957fc8722c983aa4147/clang/lib/Driver/ToolChains/Flang.cpp#L285-L288

That function in turn just forwards whatever `getCompilerRT` yields
https://github.com/llvm/llvm-project/blob/edabb5c12ed9312c3366286a69e881f625102a85/clang/lib/Driver/ToolChain.cpp#L632-L637
(though there _would_ be an aptly-named `buildCompilerRTBasename` that presumably should power `getCompilerRTBasename`).

AFAICT, this was recently changed (e.g. https://github.com/llvm/llvm-project/commit/b876596a76cdc183439b36455d26883b67f8ee51, CC @maskray) to not append the arch-suffix by default -- apparently there is a "new layout", though for some reason, the default run-times install still produces only
```
%INSTALL_PREFIX%/lib/windows/clang_rt.asan-preinit-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_cxx-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_dll_thunk-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_dynamic-x86_64.dll 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_dynamic-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_dynamic_runtime_thunk-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.asan_static-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.builtins-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.fuzzer-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.fuzzer_interceptors-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.fuzzer_no_main-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.profile-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.stats-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.stats_client-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.ubsan_standalone-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/clang_rt.ubsan_standalone_cxx-x86_64.lib 
%INSTALL_PREFIX%/lib/windows/orc_rt-x86_64.lib 
```
-- in other words, all respective libraries still have the arch suffix.

When trying to compile something with flang (built from main) on windows -- in the same way that did work with LLVM 17&18 --, this then yields:
```
LINK : fatal error LNK1104: cannot open file 'clang_rt.builtins.lib'
```

**Comment by banach-space** [comment]
@DavidTruby , would you be able to help with this?

**Comment by DavidTruby** [comment]
I've reverted the commit I believe is causing this issue, could you try again?

**Comment by h-vetinari** [comment]
OK, I presume this was https://github.com/llvm/llvm-project/commit/a2d340ba161fe48ee4ff736c6e7877038a7388cd. I'll build main on top of that and report back. Thank you!

**Comment by h-vetinari** [comment]
Can confirm that this works again now, thanks a lot!
